src/hooks/use-query.tsx

```diff
@@ -0,0 +1,41 @@
+import {
+  useQuery as useReactQuery,
+  UseQueryOptions
+} from '@tanstack/react-query'
+import { AxiosError } from 'axios'
+
+interface UseQueryProps<
+  TQueryFnData,
+  TError,
+  TData,
+  TQueryKey extends unknown[]
+> {
+  queryKey: TQueryKey
+  queryFn: () => Promise<TQueryFnData>
+  options?: UseQueryOptions<TQueryFnData, TError, TData>
+}
+
+const useQuery = <
+  TQueryFnData,
+  TError = AxiosError,
```

Comment:
I don't think that we need all those fields inside `AxiosError` object. If we will use this error inside our components, we will have to manually map this object to type that we need and take only those fields, that we need. This logic will be duplicated inside all components that are using the error object.
Maybe it would be better to extract only needed fields here (with some helper/mapper function ofc)? 
![image](https://github.com/user-attachments/assets/e1f0fb9c-007b-4dbb-820b-299d4b9b713f)